src/redprl/refiner.fun

```diff
@@ -1015,8 +1013,190 @@ struct
       end
   end
 
+  structure Restriction =
+  struct
+    (* This structure provides functions that automate the restriction
+       judgement rules given in "Dependent Cubical Realizability",
+       page 46.
+
+       Such automation is possible because the rules are "syntax
+       directed" on the restriction being performed.
+     *)
+
+    fun neqDims (P.APP P.DIM0, P.APP DIM1) = true
+      | neqDims (P.APP P.DIM1, P.APP DIM0) = true
+      | neqDims _ = false
```

Comment:
`neqDims` is used to implement the part of the restriction judgment that "finds a contradiction" such as `0 = 1`. Since `x = 0` is not a contradiction, it is important that both sides are constants, and that they are different. I'll clarify this.